Add files via upload

nashjc opened the pull request:
These have the hessian code

jlmelville commented:
Thank you for the PR @nashjc! Couple of initial questions:

1. Can you add yourself to the `DESCRIPTION` as a contributor?
2. Several of the Hessian functions have `error` mentioned in the comments. How concerning is that? Is it just for a specific evaluation point? Is it something that could be added to the documentation of the functions?

nashjc commented:
DESCRIPTION: Added ctb field. Hope I've done it right.

Removed working note crud.

nashjc commented:
Hope it's OK, JN

jlmelville commented on R/08_bard.R:
```diff
@@ -85,6 +87,43 @@ bard <- function() {
+             h[3,3] <- h[3,3] + 2.0*s2*d3 ^ 2
+             
+         } else {
+             h[1:n, 1,] <- .Machine$double.xmax
```
`n` is not defined

jlmelville commented on R/08_bard.R:
```diff
@@ -85,6 +87,43 @@ bard <- function() {
+             
+         } else {
+             h[1:n, 1,] <- .Machine$double.xmax
+             flag <- - 3
+             return
```
What is intended here? `return(h)`? How is `flag` communicated to the caller? Getting flashbacks to my Fortran77 days here :)

jlmelville commented on R/15_kow_osb.R:
```diff
@@ -90,6 +92,42 @@ kow_osb <- function() {
+             h[3,4] <- h[3,4] + 2.0*x1*u[i] ^ 2*s1/t1 ^ 3*( x1*u[i]*s1/t1 - 2.0*d1 )
+             h[4,4] <- h[4,4] + 2.0*x1*u[i]*s1/t1 ^ 3*( x1*u[i]*s1/t1 - 2.0*d1 )
+          } else {
+             h<-matrix(.Machine$double.eps, nro4, ncol=4)
+             flag <- - 3
+             return
```
Similar issue here (also `nro4` isn't defined).

jlmelville commented on R/26_trigon.R:
```diff
@@ -56,6 +58,7 @@
 #' @export
 trigon <- function() {
   list(
+    m = length(par),
```
Unfortunately `par` is not defined at this point.

jlmelville commented on R/25_var_dim.R:
```diff
@@ -47,6 +49,7 @@
 #' @export
 var_dim <- function() {
   list(
+    m = length(par) + 1,
```
same problem here with as in `trigon`: `par` is not defined yet.

jlmelville commented on R/10_meyer.R:
```diff
@@ -88,6 +90,37 @@ meyer <- function() {
+          h[2,3] <- h[2,3] - 2.0*t1*x1/d2 ^ 2*( t2 + s1*x2/d2 )
+          h[3,3] <- h[3,3] + 2.0*t1*x1*x2/d2 ^ 3*( 2.0*t2 + s1*x2/d2 )
+        } else {
+          h <- matrix(0.0, ncol=3, nrow=3)
+          flag <- - 3
+          return
```
another `return` issue.

jlmelville reviewed (changes requested):
First of all, this is a massive amount of very difficult work which I was unable to find the fortitude to even begin to attempt, so I really appreciate the immense effort which must have gone into this.

There are a few places where there are undefined variables and some syntax that I think needs some conversion (e.g. around `return`), but it's only very minor stuff. Would you mind taking a look and making those changes?

If you have `devtools` installed then `devtools::check()` should flag up these kinds of issues (or ctrl-shift-E after suitably configuring RStudio).

nashjc commented:
@jlmelville commented on this pull request.

In R/08_bard.R:

> +
+         s1 <- d2*x2 + d3*x3
+         if ( s1 != 0.0 ) {
+             t1 <- y8[i] - ( x1 + ( d1 / s1 ) )
+             t2 <- d1 / s1 ^ 2
+             t3 <- - 2.0*d1 / s1 ^ 3
+             s2 <- t1*t3 + t2 ^ 2
+             h[1,1] <- h[1,1] + 2.0
+             h[1,2] <- h[1,2] - 2.0*d2*t2
+             h[2,2] <- h[2,2] + 2.0*s2*d2 ^ 2
+             h[1,3] <- h[1,3] - 2.0*t2*d3
+             h[2,3] <- h[2,3] + 2.0*s2*d2*d3
+             h[3,3] <- h[3,3] + 2.0*s2*d3 ^ 2
+             
+         } else {
+             h[1:n, 1,] <- .Machine$double.xmax

n is not defined

JN: Now defined in code.

---------------------------------------------------------------
@jlmelville commented on this pull request.

In R/08_bard.R:

> +             flag <- - 3
+             return

What is intended here? return(h)? How is flag communicated to the caller? Getting flashbacks to my Fortran77 days here :)

JN: leftovers from the Fortran code of mgh. Commented out

---------------------------------------------------------------

@jlmelville commented on this pull request.

In R/15_kow_osb.R:

> +             h<-matrix(.Machine$double.eps, nro4, ncol=4)
+             flag <- - 3
+             return

Similar issue here (also nro4 isn't defined).

JN: commented out flag. 
    nro4 ->  nrow=4
   (typo)
---------------------------------------------------------------
@jlmelville commented on this pull request.

In R/26_trigon.R:

> @@ -56,6 +58,7 @@
 #' @export
 trigon <- function() {
   list(
+    m = length(par),

Unfortunately par is not defined at this point.

JN: replaced
    m = 30

---------------------------------------------------------------
@jlmelville commented on this pull request.

In R/25_var_dim.R:

> @@ -47,6 +49,7 @@
 #' @export
 var_dim <- function() {
   list(
+    m = length(par) + 1,

same problem here with as in trigon: par is not defined yet.

JN: removed. Wasn't needed.
---------------------------------------------------------------
@jlmelville commented on this pull request.

In R/10_meyer.R:

> +          h <- matrix(0.0, ncol=3, nrow=3)
+          flag <- - 3
+          return

another return issue.

JN: there were three returns needing "()" added.
    flag commented
---------------------------------------------------------------

jlmelville commented:
@nashjc was this closed by accident? I was about to say it looked good to me and merge it. Let me know if it's good to go.

jlmelville commented:
Many thanks for this @nashjc! This should all be merged into the `master` branch now.
